ebmbot/slack.py

```diff
@@ -46,10 +51,27 @@ def notify_slack(
             sleep(1)
             error = err
 
-    logger.error(
-        "Could not notify slack",
-        channel=channel,
-        message=message_text,
-        thread_ts=thread_ts,
-        error=error,
+    msg_kwargs["text"] = COULD_NOT_NOTIFY_SLACK
+    msg_kwargs["blocks"] = get_slack_error_blocks(message_text, error)
+    try:
+        slack_client.chat_postMessage(**msg_kwargs)
+    except Exception:
+        logger.error(
+            COULD_NOT_NOTIFY_SLACK,
+            channel=channel,
+            message=message_text,
+            thread_ts=thread_ts,
+            error=error,
+        )
+
+
+def get_slack_error_blocks(message_text, error):
+    return get_basic_header_and_text_blocks(
+        header_text=COULD_NOT_NOTIFY_SLACK,
+        texts=[
+            "Slack encountered the error",
+            f"```{str(error)}```",
+            "when trying to post the following message:",
+            f"```{message_text}```",
```

Comment:
There is a risk here that this'll hit slack's limits on blocks; there's a max blocks in a message (50) which it won't encounter, but also a max characters per text block (3000). `funding_report.py` does some handling for the max number of blocks, just by truncating them. If we're posting arbitrary message text as a block, we might want to truncate that too. 

The simplest way would be to handle it here and just send `f"```{message_text:2994}```"`.

Or I guess maybe we can check it first and send with an ellipsis to indicate it's been truncated:
```
if len(message_text) > 2991:
    message_text = f"```{message_text:2991}...```"
```
(I'm not convinced my maths is correct there :slightly_smiling_face:)

Reply:
Implemented the version with the check :)